### ISSUE TITLE
linter: false positive in `no-this-before-super`

### ISSUE THREAD

**Issue by rzvxa** [oxc-project/oxc#3668]
https://github.com/rzvxa/oxlint-ecosystem-ci/actions/runs/9511509383/job/26217870705#step:5:66

eg:

```js
function f() {
    try {
        return a();
    }
    catch (err) {
        throw new class CustomError extends Error {
            constructor() {
                super(err);
            }
        };
    }
    finally {
        this.b();
    }
}
```
